### DISCUSSION TITLE
Node.js remote package is missing?

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#9996] [CLI - Electron mode] [answered]
In previous versions of Quasar i was using the following code in my utils.js file...

```
const { remote } = require('electron')

/**
 * Prompts user with dialog to select config file
 */
export function selectConfigFile(): string {
  const mainWindow = remote.getCurrentWindow();
  let filepaths = remote.dialog.showOpenDialogSync(mainWindow, {
    defaultPath: '/home',
    filters: [
      { name: 'All Files', extensions: ['*'] },
      { name: 'Filter Preset', extensions: ['json'] }
    ],
    properties: ['openFile']
  });
  if (!filepaths) {
    return '';
  }
  return filepaths[0];
};
```

I've moved the following code into a electron-preload.js file but still get errors.
How do I resolve the following error... 
```
runtime-core.esm-bundler.js?9e79:217 Uncaught Error: remote.getCurrentWindow is not a function
    at selectConfigFile (SettingsPanel.vue?6bf9:118)
    at callWithErrorHandling (runtime-core.esm-bundler.js?9e79:154)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js?9e79:163)
    at HTMLElement.invoker (runtime-dom.esm-bundler.js?8886:367)
```

**Comment by JokerMartini** [comment]
I've tried a variety of options and I can't seem to get either of these to work as they say remote does not exist. Anyone else exp this? 

const mainWindow = remote.getCurrentWindow();
filepaths = remote.dialog.showOpenDialogSync(...

**Comment by rstoenescu** [comment, marked as answer]
Please be aware of the Electron version that you are using and then plan your code accordingly. "This and that" not being available is not due to Quasar, but exclusively due to newer Electron. Use the electron-preload script to bridge between nodejs and your UI thread. https://quasar.dev/quasar-cli/developing-electron-apps/electron-upgrade-guide#the-electron-preload-js-file

Also notice this in /src-electron/electron-main.js:
```
new BrowserWindow({
    // ...
    webPreferences: { // <<<<< this is really important
    }
}
```

Read about how you can enable different Electron stuff (including the remote object if I'm not mistaken) through `webPreferences`. But the defaults really have a reason to be set the way they are. Make sure that you won't blow up your app security-wise by enabling different stuff.

Regardless, googling about the Electron webPreference or looking directly into the [Electron docs](https://www.electronjs.org/docs) will untangle any "issues" you might have.
